### PR TITLE
GCG-6 trigger codegen with parameters, powerd by flask

### PR DESCRIPTION
**Pull request by prncoprs**
feature finished：
1. receive a json file in a web page
2. when upload the json file, the page return the result of "java -jar codegen.jar <uploaded_json_file> <generated_jar_path>"
3. remove the unnecessary files
4. add a function to test if a file is a json file

### CODE REVIEW

**Review by DanishAlsayed — COMMENTED** [review]
How can the functionality added in this PR be tested in an automated fashion? Are there python/flask testing frameworks you can use?

**Comment by prncoprs** [comment]
Add the unit test in the new test.py file. 
For now, the unittest only test function `is_json_file()`, the unittest of upload function has not yet finished.

**Review by DanishAlsayed — COMMENTED** [review]
> Add the unit test in the new test.py file.
> For now, the unittest only test function `is_json_file()`, the unittest of upload function has not yet finished.

Great, please add as many unit tests as possible and have a look at my comment on `gui/flask/test.py`

File: gui/flask/test.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        fake_json_file_content = {'flink':1, 'spark':2, 'streaming':"yes"}
+        fake_json_file_path = "./fake_json_file.json"
+        with open(fake_json_file_path, 'w') as f:
+            json.dump(fake_json_file_content, f, indent=4)
+        
+        # test the function is_json_file()
+        result = main.is_json_file(fake_json_file_path)
+        
+        # delete the fake json file
+        if os.path.exists(fake_json_file_path):
```
[the remark this answers is not in the document]
Comment (prncoprs) [reply gui/flask/test.py:24]: I added `setUp` and `tearDown` functions to prepare and drop the material that are needed in the test case.
And I added the positive and negative test case to validate a json file.

**Comment by prncoprs** [comment]
Finished both upload and json validation unittest.
Submitting the generated jar to Flink and beautifying the web page are under development.

**Comment by prncoprs** [comment]
Finshed basic run task in flink and the relevant unittest.

**Comment by DanishAlsayed** [comment]
Move test resources to "resources" folder under "tests"

**Comment by DanishAlsayed** [comment]
rename the folder from "flask_project" to "AJUFlask"

**Comment by DanishAlsayed** [comment]
removed commented code and any unused files/images etc.

**Comment by DanishAlsayed** [comment]
Use absolute paths as argument to codegen

**Comment by DanishAlsayed** [comment]
double check test coverage

**Comment by DanishAlsayed** [comment]
check if process is running instead of checking for a string that the codegen outputs

**Comment by prncoprs** [comment]
@DanishAlsayed 
For now, it's ready to merge GCG-6 to master.
Finished:
1. move test resources to "resources" folder under "tests"
2. rename the folder from "flask_project" to "aju_flask"
3. removed the unnecessary comments and files
4. make all cofigure path absolute path
5. replace the return of `run_flink_task()` from string to a class, which contains a return code
6. the code coverage is as below:

Code Coverage：
Name     Stmts   Miss  Cover
----------------------------
run.py      59      2    97%

**Review by wqc0712 — APPROVED** [review]
The update on codegen should not be included in the branch related to gui.  @DanishAlsayed Please make sure the codegen/pom.xml is correct when you merge your next branch.

**Pull request merged**